File: CSharp/TestClient/XPayloadGenerator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

using UsingMiddleware;

namespace TestClient
{
    public class XPayloadGenerator
    {
        private static readonly Random _random = new Random(42); 
        private static int _personIdCounter = 1;

        public static XPayload Create3KBPayload()
        {
            var payload = new XPayload
            {
                ByteField = (byte)_random.Next(0, 255),
                SByteField = (sbyte)_random.Next(-128, 127),
                ShortField = (short)_random.Next(-32768, 32767),
                UShortField = (ushort)_random.Next(0, 65535),
                IntField = _random.Next(),
                UIntField = (uint)_random.Next(),
                LongField = _random.NextInt64(),
                ULongField = (ulong)_random.NextInt64() & 0x7FFFFFFFFFFFFFFF,

                FloatField = (float)(_random.NextDouble() * 1000),
                DoubleField = _random.NextDouble() * 10000,

                DecimalField = (decimal)(_random.NextDouble() * 100000),

                BoolField = _random.Next(2) == 1,

                CharField = (char)_random.Next(65, 90), 
                StringField = GenerateRandomString(200), 

                GuidField = Guid.NewGuid(),

                DateTimeField = DateTime.UtcNow.AddDays(_random.Next(-365, 365)),
                TimeSpanField = TimeSpan.FromMinutes(_random.Next(0, 1440)), // 0-24 horas

                Persons = GeneratePersonList(8) 
            };

            var options = new JsonSerializerOptions
            {
                WriteIndented = false, 
                PropertyNamingPolicy = JsonNamingPolicy.CamelCase
            };

            return payload;
        }

        private static List<XPersonData> GeneratePersonList(int count)
        {
            var persons = new List<XPersonData>();

            for (int i = 0; i < count; i++)
            {
                persons.Add(new XPersonData
                {
                    Id = _personIdCounter++,
                    PersonGuid = Guid.NewGuid(),
                    FirstName = GenerateFirstName(),
                    LastName = GenerateLastName(),
                    Age = _random.Next(18, 80),
                    BirthDate = DateTime.UtcNow.AddYears(-_random.Next(18, 80)).AddDays(_random.Next(0, 365)),
                    Email = GenerateEmail(),
                    PhoneNumber = GeneratePhoneNumber(),
                    Address = GenerateAddress(),
                    IsActive = _random.Next(100) < 95 // 95% de ativos
                });
            }

            return persons;
        }

        private static XAddressInfo GenerateAddress()
        {
            return new XAddressInfo
            {
                Street = GenerateStreetName(),
                Number = _random.Next(1, 5000).ToString(),
                Complement = _random.Next(3) == 0 ? GenerateRandomString(10) : null,
                Neighborhood = GenerateNeighborhood(),
                City = GenerateCity(),
                State = GenerateState(),
                PostalCode = GeneratePostalCode(),
                Country = "Brasil",
                Latitude = _random.NextDouble() * 180 - 90,
                Longitude = _random.NextDouble() * 360 - 180
            };
        }

        private static string GenerateRandomString(int length)
        {
            const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789 áéíóúàèìòùâêîôûãõç";
            var stringChars = new char[length];

            for (int i = 0; i < length; i++)
            {
                stringChars[i] = chars[_random.Next(chars.Length)];
            }

            return new string(stringChars);
        }

        private static string GenerateFirstName()
        {
            var firstNames = new[]
            {
            "Maria", "João", "Ana", "Pedro", "Carlos", "Fernanda", "Roberto", "Juliana",
            "Marcos", "Patrícia", "Alexandre", "Camila", "Ricardo", "Beatriz", "Gustavo",
            "Larissa", "Felipe", "Amanda", "Daniel", "Carolina", "Thiago", "Renata",
            "Vinícius", "Tatiane", "Eduardo", "Vanessa", "Anderson", "Priscila",
            "Leonardo", "Débora", "Marcelo", "Simone", "Rafael", "Cristina"
        };

            return firstNames[_random.Next(firstNames.Length)];
        }

        private static string GenerateLastName()
        {
            var lastNames = new[]
            {
            "Silva", "Santos", "Oliveira", "Souza", "Pereira", "Costa", "Carvalho",
            "Almeida", "Ferreira", "Ribeiro", "Alves", "Monteiro", "Barbosa",
            "Lima", "Mendes", "Freitas", "Barros", "Pinto", "Vieira", "Araújo",
            "Rocha", "Martins", "Rodrigues", "Nascimento", "Campos", "Machado",
            "Cardoso", "Teixeira", "Medeiros", "Nunes", "Moraes", "Machado"
        };

            return lastNames[_random.Next(lastNames.Length)];
        }

        private static string GenerateEmail()
        {
            var domains = new[] { "gmail.com", "hotmail.com", "yahoo.com", "outlook.com", "empresa.com.br" };
            var firstName = GenerateFirstName().ToLower().Replace(" ", ".");
            var lastName = GenerateLastName().ToLower().Replace(" ", ".");
            var domain = domains[_random.Next(domains.Length)];

            return $"{firstName}.{lastName}{_random.Next(1, 999)}@{domain}";
        }

        private static string GeneratePhoneNumber()
        {
            return $"({_random.Next(11, 99)}) 9{_random.Next(1000, 9999)}-{_random.Next(1000, 9999)}";
        }

        private static string GenerateStreetName()
        {
            var streets = new[]
            {
            "Rua das Flores", "Avenida Brasil", "Rua dos Andradas", "Avenida Paulista",
            "Rua Oscar Freire", "Rua Augusta", "Avenida Rio Branco", "Rua XV de Novembro",
            "Avenida Atlântica", "Rua da Consolação", "Avenida Brigadeiro Faria Lima",
            "Rua Bela Vista", "Avenida Copacabana", "Rua Direita", "Avenida Europa"
        };

            return streets[_random.Next(streets.Length)];
        }

        private static string GenerateNeighborhood()
        {
            var neighborhoods = new[]
            {
            "Centro", "Jardins", "Moema", "Vila Madalena", "Pinheiros", "Ipanema",
            "Copacabana", "Leblon", "Botafogo", "Flamengo", "Lapa", "Santa Teresa",
            "Barra da Tijuca", "Recreio", "Tijuca", "Méier", "Copacabana", "Botafogo"
        };

            return neighborhoods[_random.Next(neighborhoods.Length)];
        }

        private static string GenerateCity()
        {
            var cities = new[]
            {
            "São Paulo", "Rio de Janeiro", "Belo Horizonte", "Porto Alegre",
            "Curitiba", "Salvador", "Fortaleza", "Brasília", "Recife", "Manaus"
        };

            return cities[_random.Next(cities.Length)];
        }

        private static string GenerateState()
        {
            var states = new[]
            {
            "SP", "RJ", "MG", "RS", "PR", "BA", "CE", "DF", "PE", "AM"
        };

            return states[_random.Next(states.Length)];
        }

        private static string GeneratePostalCode()
        {
            return $"{_random.Next(10000, 99999)}-{_random.Next(100, 999)}";
        }

        public static List<XPayload> CreateMultiplePayloads(int count)
        {
            var payloads = new List<XPayload>();
            for (int i = 0; i < count; i++)
            {
                payloads.Add(Create3KBPayload());
            }
            return payloads;
        }
    }
}

File: Rust/TestClient/src/XPayloadGenerator.rs
use std::sync::atomic::{AtomicI32, Ordering};
use std::time::{SystemTime};
use rand::{Rng, SeedableRng};
use rand::rngs::StdRng;
use crate::XPayload::{XAddressInfo, XPayload, XPersonData};

pub struct XPayloadGenerator;

static PERSON_ID_COUNTER: AtomicI32 = AtomicI32::new(1);

impl XPayloadGenerator {
    pub fn create_3kb_payload() -> XPayload {
        let mut rng = StdRng::seed_from_u64(42);

        let payload = XPayload {
            ByteField: rng.random::<u8>(),
            SByteField: rng.random::<i8>(),
            ShortField: rng.random::<i16>(),
            UShortField: rng.random::<u16>(),
            IntField: rng.random::<i32>(),
            UIntField: rng.random::<u32>(),
            LongField: rng.random::<i64>(),
            ULongField: rng.random::<u64>() >> 1,

            FloatField: rng.random::<f32>() * 1000.0,
            DoubleField: rng.random::<f64>() * 10000.0,
            DecimalField: (rng.random::<f64>() * 100000.0) as f64,

            BoolField: rng.random::<bool>(),

            CharField: XPayloadGenerator::generate_random_char(&mut rng),
            StringField: XPayloadGenerator::generate_random_string(&mut rng, 200),

            GuidField: uuid::Uuid::new_v4(),

            // DateTimeField: SystemTime::now().into(),
            // TimeSpanField: Duration::from_secs(rng.random_range(0..1440*60)),

            Persons: XPayloadGenerator::generate_person_list(&mut rng, 8),
        };

        payload
    }

    fn generate_person_list(rng: &mut StdRng, count: usize) -> Vec<XPersonData> {
        let mut persons = Vec::with_capacity(count);

        for _ in 0..count {
            let id = PERSON_ID_COUNTER.fetch_add(1, Ordering::SeqCst);
             let birth_date = SystemTime::now().into();
            persons.push(XPersonData {
                Id: id,
                PersonGuid: uuid::Uuid::new_v4(),
                FirstName: XPayloadGenerator::generate_first_name(rng),
                LastName: XPayloadGenerator::generate_last_name(rng),
                Age: rng.random_range(18..80),
                 BirthDate: birth_date,
                Email: XPayloadGenerator::generate_email(rng),
                PhoneNumber: XPayloadGenerator::generate_phone_number(rng),
                Address: XPayloadGenerator::generate_address(rng),
                IsActive: rng.random_range(0..100) < 95,
            });
        }

        persons
    }

    fn generate_address(rng: &mut StdRng) -> XAddressInfo {
        XAddressInfo {
            Street: XPayloadGenerator::generate_street_name(rng),
            Number: rng.random_range(1..5000).to_string(),
            Complement: if rng.random_range(0..3) == 0 {
                XPayloadGenerator::generate_random_string(rng, 10)
            } else {
                String::new()
            },
            Neighborhood: XPayloadGenerator::generate_neighborhood(rng),
            City: XPayloadGenerator::generate_city(rng),
            State: XPayloadGenerator::generate_state(rng),
            PostalCode: XPayloadGenerator::generate_postal_code(rng),
            Country: "Brasil".to_string(),
            Latitude: rng.random_range(-90.0..90.0),
            Longitude: rng.random_range(-180.0..180.0),
        }
    }

    fn generate_random_string(rng: &mut StdRng, length: usize) -> String {
        const CHARSET: &[u8] = b"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789 ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
        (0..length)
            .map(|_| {
                let idx = rng.random_range(0..CHARSET.len());
                CHARSET[idx] as char
            })
            .collect()
    }

    fn generate_random_char(rng: &mut StdRng) -> char {
        (rng.random_range(65..91) as u8) as char
    }

    fn generate_first_name(rng: &mut StdRng) -> String {
        let list = vec!["Maria", "João", "Ana", "Pedro", "Carlos", "Fernanda", "Roberto", "Juliana", "Marcos", "Patrícia"];
        list[rng.random_range(0..list.len())].to_string()
    }

    fn generate_last_name(rng: &mut StdRng) -> String {
        let list = vec!["Silva", "Santos", "Oliveira", "Souza", "Pereira", "Costa"];
        list[rng.random_range(0..list.len())].to_string()
    }

    fn generate_email(rng: &mut StdRng) -> String {
        let domains = vec!["gmail.com", "hotmail.com", "yahoo.com", "outlook.com", "empresa.com.br"];
        format!(
            "{}.{}{}@{}",
            XPayloadGenerator::generate_first_name(rng).to_lowercase(),
            XPayloadGenerator::generate_last_name(rng).to_lowercase(),
            rng.random_range(1..999),
            domains[rng.random_range(0..domains.len())]
        )
    }

    fn generate_phone_number(rng: &mut StdRng) -> String {
        format!("({}) 9{}-{}", rng.random_range(11..99), rng.random_range(1000..9999), rng.random_range(1000..9999))
    }

    fn generate_street_name(rng: &mut StdRng) -> String {
        let streets = vec!["Rua das Flores", "Avenida Brasil", "Rua dos Andradas"];
        streets[rng.random_range(0..streets.len())].to_string()
    }

    fn generate_neighborhood(rng: &mut StdRng) -> String {
        let list = vec!["Centro", "Jardins", "Moema", "Vila Madalena"];
        list[rng.random_range(0..list.len())].to_string()
    }

    fn generate_city(rng: &mut StdRng) -> String {
        let list = vec!["São Paulo", "Rio de Janeiro", "Belo Horizonte"];
        list[rng.random_range(0..list.len())].to_string()
    }

    fn generate_state(rng: &mut StdRng) -> String {
        let list = vec!["SP", "RJ", "MG"];
        list[rng.random_range(0..list.len())].to_string()
    }

    fn generate_postal_code(rng: &mut StdRng) -> String {
        format!("{}-{}", rng.random_range(10000..99999), rng.random_range(100..999))
    }

    pub fn create_multiple_payloads(count: usize) -> Vec<XPayload> {
        (0..count).map(|_| Self::create_3kb_payload()).collect()
    }
}
